Library/Homebrew/download_strategy.rb

```diff
@@ -282,6 +283,14 @@ def initialize(name, resource)
   end
 
   def fetch
+    if ENV["HOMEBREW_ARTIFACT_DOMAIN"]
+      if @url.include?(BottleSpecification::DEFAULT_DOMAIN)
```

Comment:
Better yet: `@url.start_with?(BottleSpecification::DEFAULT_DOMAIN)`. Or are we expecting to replace this in the middle of the URL string?